fm-workbench/simulator/edu.uah.rsesc.aadlsimulator.ui/src/edu/uah/rsesc/aadlsimulator/ui/launch/SimulationTab.java

```diff
@@ -66,114 +67,111 @@ and documentation (collectively "the Data"), to deal in the Data without restric
-		projectTxt.addModifyListener(new ModifyListener() {
-			@Override
-			public void modifyText(final ModifyEvent e) {
-				updateLaunchConfigurationDialog();
-			}			
-		});
-		
+		projectTxt.addModifyListener(e -> updateLaunchConfigurationDialog());
+
 		final Button projectSelectBtn = createPushButton(root, "Select...", null);
 		projectSelectBtn.addSelectionListener(new SelectionAdapter() {
 			@Override
-			public void widgetSelected(final SelectionEvent e) {				
-				// Show selection dialog				
+			public void widgetSelected(final SelectionEvent e) {
+				// Show selection dialog
 				final ElementListSelectionDialog dlg = new ElementListSelectionDialog(PlatformUI.getWorkbench().getActiveWorkbenchWindow().getShell(), new LabelProvider() {
 					@Override
 					public String getText(final Object element) {
 						if(element instanceof IProject){
 							return ((IProject)element).getName();
 						} else {
 							return element.toString();
-						}				
+						}
 					}
 				});
-				
+
 				dlg.setTitle("Project Selection");
 				dlg.setMessage("Select the project containing the component implementation to simulate.");
 				dlg.setElements(ResourcesPlugin.getWorkspace().getRoot().getProjects());
-				if(dlg.open() == Dialog.CANCEL) {
+				if(dlg.open() == Window.CANCEL) {
 					return;
 				}
-				
+
 				final Object selection = dlg.getFirstResult();
 				if(selection instanceof IProject) {
 					projectTxt.setText(((IProject)selection).getName());
 				}
 			}
 		});
-		
+
 		final Label componentImplementationLbl = new Label(root, SWT.NONE);
 		componentImplementationLbl.setText("Component Implementation:");
 		componentImplementationTxt = new Text(root, SWT.SINGLE | SWT.BORDER);
 		GridDataFactory.fillDefaults().grab(true, false).hint(200, SWT.DEFAULT).applyTo(componentImplementationTxt);
-		componentImplementationTxt.addModifyListener(new ModifyListener() {
-			@Override
-			public void modifyText(final ModifyEvent e) {
-				updateLaunchConfigurationDialog();
-			}			
-		});
-		
+		componentImplementationTxt.addModifyListener(e -> updateLaunchConfigurationDialog());
+
 		final Button componentImplSelectBtn = createPushButton(root, "Select...", null);
 		componentImplSelectBtn.addSelectionListener(new SelectionAdapter() {
 			@Override
 			public void widgetSelected(final SelectionEvent e) {
 				// Get list of component implementations
 				final List<IEObjectDescription> componentImplementationDescriptions = new ArrayList<IEObjectDescription>();
+				Injector injector = IResourceServiceProvider.Registry.INSTANCE
+						.getResourceServiceProvider(URI.createFileURI("dummy.aadl")).get(Injector.class);
+				final ResourceDescriptionsProvider resourceDescriptionsProvider = injector
+						.getInstance(ResourceDescriptionsProvider.class);
+				final IResourceDescriptions resourceDescriptions = resourceDescriptionsProvider
+						.getResourceDescriptions(OsateResourceUtil.getResourceSet());
```

Comment:
better just use `new ResourceSet()`. We'll likely deprecate `OsateResourceUtil`.